src/scrapers/hapoalim.js

```diff
@@ -27,34 +27,38 @@ function convertTransactions(txns) {
 async function fetchAccountData(page, options) {
   const apiSiteUrl = `${BASE_URL}/ServerServices`;
   const accountDataUrl = `${apiSiteUrl}/general/accounts`;
-  const accountInfo = await fetchGetWithinPage(page, accountDataUrl);
-  const accountNumber = `${accountInfo[0].bankNumber}-${accountInfo[0].branchNumber}-${accountInfo[0].accountNumber}`;
+  const accountsInfo = await fetchGetWithinPage(page, accountDataUrl);
 
-  const defaultStartMoment = moment().subtract(1, 'years').add(1, 'day');
-  const startDate = options.startDate || defaultStartMoment.toDate();
-  const startMoment = moment.max(defaultStartMoment, moment(startDate));
+  const accounts = [];
+  for (let accountIndex = 0; accountIndex < accountsInfo.length; accountIndex += 1) {
+    const accountNumber = `${accountsInfo[accountIndex].bankNumber}-${accountsInfo[accountIndex].branchNumber}-${accountsInfo[accountIndex].accountNumber}`;
 
-  const startDateStr = startMoment.format(DATE_FORMAT);
-  const endDateStr = moment().format(DATE_FORMAT);
-  const txnsUrl = `${apiSiteUrl}/current-account/transactions?accountId=${accountNumber}&numItemsPerPage=150&retrievalEndDate=${endDateStr}&retrievalStartDate=${startDateStr}&sortCode=1`;
+    const defaultStartMoment = moment().subtract(1, 'years').add(1, 'day');
+    const startDate = options.startDate || defaultStartMoment.toDate();
+    const startMoment = moment.max(defaultStartMoment, moment(startDate));
 
-  const txnsResult = await fetchGetWithinPage(page, txnsUrl);
+    const startDateStr = startMoment.format(DATE_FORMAT);
+    const endDateStr = moment().format(DATE_FORMAT);
```

Comment:
@chilikla I believe `startDateStr` and `endDateStr` can be moved up, outside of the for loop block.